MediaBrowser.Common/Providers/ProviderIdParsers.cs

```diff
@@ -0,0 +1,125 @@
+﻿#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace MediaBrowser.Common.Providers
+{
+    /// <summary>
+    /// Parsers for provider ids.
+    /// </summary>
+    public static class ProviderIdParsers
+    {
+        private const int ImdbMinNumbers = 7;
+        private const int ImdbMaxNumbers = 8;
+
+        /// <summary>
+        /// Parses an IMDb id from a string.
+        /// </summary>
+        /// <param name="text">The text to parse.</param>
+        /// <param name="imdbId">The parsed IMDb id.</param>
+        /// <returns>True if parsing was successful, false otherwise.</returns>
+        public static bool TryParseImdbId(ReadOnlySpan<char> text, [NotNullWhen(true)] out string? imdbId)
+        {
+            var tt = "tt".AsSpan();
+
+            // imdb id is at least 9 chars (tt + 7 numbers)
+            while (text.Length >= 2 + ImdbMinNumbers)
+            {
+                var ttPos = text.IndexOf(tt);
+                if (ttPos == -1)
+                {
+                    imdbId = default;
+                    return false;
+                }
+
+                text = text.Slice(ttPos + tt.Length);
+                var i = 0;
+                for (; i < Math.Min(text.Length, ImdbMaxNumbers); i++)
+                {
+                    var c = text[i];
+                    if (!IsDigit(c))
+                    {
+                        break;
+                    }
+                }
+
+                // skip if more than 8 digits
+                if (i <= ImdbMaxNumbers && i >= ImdbMinNumbers)
+                {
+                    imdbId = string.Concat(tt, text.Slice(0, i));
```

Comment:
No reason to do a concat here, you should just return a `ReadOnlySpan` starting from `ttPos` with length `2 + i`

Reply:
`text` does no longer contain `tt` so I can't slice starting from `ttPos`. Is it better to copy the Span and slice the copy than `string.Concat?`